src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1169 @@
+    logging,
+)
+from .configuration_gptsan_japanese import GPTSANJapaneseConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "GPTSANJapaneseConfig"
+_TOKENIZER_FOR_DOC = "GPTNeoXJapaneseTokenizer"
+_CHECKPOINT_FOR_DOC = "Tanrei/GPTSAN-japanese"
+
+####################################################
+# This dict contains ids and associated url
+# for the pretrained weights provided with the models
+####################################################
+GPTSAN_JAPANESE_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "Tanrei/GPTSAN-japanese",
+    # See all GPTSAN-japanese models at https://huggingface.co/models?filter=gptsan-japanese
+]
+
+
+class GPTSANJapaneseNorm(nn.Module):
+    """
+    Layer Normalyzation for GPTSAN
+
+    """
+
+    def __init__(self, config: GPTSANJapaneseConfig):
+        super().__init__()
+        self.weight = nn.Parameter(torch.zeros(config.d_model))
+        self.bias = nn.Parameter(torch.zeros(config.d_model))
+        self.epsilon = config.layer_norm_epsilon
+
+    def forward(self, x):
+        r"""
+        Args:
+            hidden_states (`torch.Tensor`) :
+                [num_groups, tokens_per_group, hidden_dim] inputs to send to experts.
+        Returns:
+            torch.Tensor[num_groups, tokens_per_group, hidden_dim]
+
+        """
+        x -= torch.mean(x, dim=-1, keepdims=True)
+        s = torch.mean(x**2, dim=-1, keepdims=True)
+        x *= torch.rsqrt(s + self.epsilon)
+        return x * self.weight + self.bias
+
+
+class GPTSANJapaneseDenseActDense(nn.Module):
+    """
+    FFN Layer for Switch Transformer and Extra layers
+
+    GPTSAN can mix Switch Transformer layers and normal Transformer layers This class is used as Expert in Switch
+    Transformer layers and as FFN in regular Transformer layers. RELU is used in the Switch Transformer layer, and
+    Swish is used in the normal Transformer layer, so there is a choice of which is used in the argument.
+
+    """
+
+    def __init__(self, config: GPTSANJapaneseConfig, ext_layer=False):
+        super().__init__()
+        d_inter = config.d_ext if ext_layer else config.d_ff
+        self.wi = nn.Linear(config.d_model, d_inter, bias=ext_layer)
+        self.wo = nn.Linear(d_inter, config.d_model, bias=ext_layer)
+        self.dropout = nn.Identity() if ext_layer else nn.Dropout(config.dropout_rate)
+        self.act = ACT2FN["swish" if ext_layer else "relu"]
+
+    def forward(self, hidden_states):
+        r"""
+        Args:
+            hidden_states (`torch.Tensor`) :
+                [num_groups, tokens_per_group, hidden_dim] inputs to send to experts.
+        Returns:
+            torch.Tensor[num_groups, tokens_per_group, hidden_dim]
+
+        """
+        hidden_states = self.wi(hidden_states)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.wo(hidden_states)
+        return hidden_states
+
+
+class GPTSANJapaneseTop1Router(nn.Module):
```

Comment:
It's a little different because of ```if self.training and self.jitter_noise > 0:``` line, so commit test didn't pass with ```copied from statement```.
However, since we can set jitter_noise to 0, so I respect the original and restore it, adding ```copied from```.